splash/qtrender_lua.py

```diff
@@ -234,8 +251,15 @@ def exceptions_as_return_values_wrapper(self, *args, **kwargs):
                 res = res.result
             else:
                 res = (b'return',) + ensure_tuple(res)
+            if error_as_flag and (result_as_flag or res[1] is None or res[1] is False):
+                res = add_flag(res, True)
         except Exception as e:
-            res = (b'raise', repr(e))
+            if error_as_flag and self.FLAG_EXCEPTIONS is not None \
+                    and any(isinstance(e, x) for x in self.FLAG_EXCEPTIONS):
+                res = (b'return', False, repr(e).encode('utf-8'))
+            else:
+                res = (b'raise', repr(e))
```

Comment:
it looks like sometimes exception repr is passed as bytes (`.encode('utf8')`), and sometimes as a str (`repr(e)`) - why is that?